lib/loader.js

```diff
@@ -16,10 +16,14 @@ module.exports = function(content, map) {
 	var root = query.root;
 	var moduleMode = query.modules || query.module;
 	var camelCaseKeys = query.camelCase || query.camelcase;
+	var sourceMap = query.sourceMap || false;
 	var resolve = createResolver(query.alias);
 
-	if(map !== null && typeof map !== "string") {
+	if(sourceMap && map !== null && typeof map !== "string") {
 		map = JSON.stringify(map);
+	} else {
+        // Some loaders (example `"postcss-loader": "1.x.x"`) always generates source map, we should remove it
+		map = null;
```

Comment:
We should fix it in `postcss-loader` aswell , I spoke with andrey a while ago about this, but this was added via PR for Angular* folks, if we have clarification of souremaps 'do && don'ts'  I will make sure it works across the whole CSS Pipeline 😛 . 

* They wanted inline sourcemaps because of the angular component system (`{String}` only)

There is a PR open at `style-loader`  [#165](https://github.com/webpack-contrib/style-loader/pull/165) aswell, which would enable inline sourcemaps support in `style-loader`, the question is if that should be working in general or not and imho whatever case is the better should be 'more strictly ' followed to avoid confusion about it. I closed a few PR's in regard of sourcrmaps already, there is definitely room for UX improvements :)